### PR TITLE
[DEPENDENCIES] Upgrade to Django 4.2

### PR DESCRIPTION
**Pull request by ELDiablO59152**
### Changed
- [DEPENDENCIES] Upgrade to Django 4.2
### Fixed
- [SETTINGS] Update CSRF_TRUSTED_ORIGINS format for Django 4
- [URLS] Simplification of different paths
- [DEPRECATION] Change request.META to request.headers
- [DEPRECATION] get_random_string now requires length argument
- [DEPRECATION] Keyword "trans" replaced by "translate" in templates
- [DEPRECATION] make_aware function changed to django aware datetime

### CODE REVIEW

File: vulture_os/gui/urls.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 
     path('collapse', collapse, name="gui.collapse_menu"),
 
-    re_path(r'^$', dashboard.dashboard_services, name="gui.dashboard.services"),
+    path('', dashboard.dashboard_services, name="gui.dashboard.services"),
```
Comment (frikilax) [inline vulture_os/gui/urls.py:40]: not sure I want to make this change

File: vulture_os/gui/urls.py [hunk 2]
```diff
@@ -49,6 +49,6 @@
     re_path('^api/ci/get/(?P<objclass>[A-Za-z0-9\._]+)/(?P<object_id>[A-Fa-f0-9]+)?$',
             ApiWrapperGet.as_view(), name='gui.views.api_wrapper_get'),
 
-    re_path('^api/v1/services/monitor/$', api_view.services_monitor, name="api.services_monitor"),
+    path('api/v1/services/monitor/', api_view.services_monitor, name="api.services_monitor"),
```
Comment (frikilax) [inline vulture_os/gui/urls.py:52]: is it really necessary ?

File: vulture_os/portal/settings.py [hunk 3]
```diff
@@ -114,10 +114,9 @@
 LANGUAGE_CODE = 'en-us'
 TIME_ZONE = 'UTC'
 USE_I18N = True
-USE_L10N = True
```
Comment (frikilax) [inline vulture_os/portal/settings.py:117]: Even if enabled by default, it's better to keep explicitely setting the parameter

File: vulture_os/portal/settings.py [hunk 4]
```diff
@@ -114,10 +114,9 @@
...
 USE_TZ = True
 
-STATIC_URL = '/static/'
+STATIC_URL = 'static/'
```
Comment (frikilax) [inline vulture_os/portal/settings.py:119]: is it really necessary?

File: vulture_os/vulture_os/settings.py [hunk 5]
```diff
@@ -182,20 +182,19 @@
 
 
 # Internationalization
-# https://docs.djangoproject.com/en/2.0/topics/i18n/
+# https://docs.djangoproject.com/en/4.2/topics/i18n/
 
 LANGUAGE_CODE = 'en-us'
 TIME_ZONE = 'UTC'
 USE_I18N = True
-USE_L10N = True
```
Comment (frikilax) [inline vulture_os/vulture_os/settings.py:191]: same question: why is it necessary to remove this line?

File: vulture_os/vulture_os/settings.py [hunk 6]
```diff
@@ -182,20 +182,19 @@
...
 
 
 # Static files (CSS, JavaScript, Images)
-# https://docs.djangoproject.com/en/2.0/howto/static-files/
-MEDIA_PATH = '/gui/static/img/'
+# https://docs.djangoproject.com/en/4.2/howto/static-files/
+MEDIA_PATH = 'gui/static/img/'
 STATIC_ROOT = os.path.join(BASE_DIR, "static")
-STATIC_URL = '/static/'
+STATIC_URL = 'static/'
```
Comment (frikilax) [inline vulture_os/vulture_os/settings.py:197]: is it also necessary?

File: vulture_os/portal/settings.py (hunk 3 above)
Comment (ELDiablO59152) [reply vulture_os/portal/settings.py:117]: It is deprecated and throw a warning on django check routine : https://github.com/adamchainz/django-upgrade#use_l10n

File: vulture_os/portal/settings.py (hunk 4 above)
Comment (ELDiablO59152) [reply vulture_os/portal/settings.py:119]:
```
To allow serving a Django site on a subpath without changing the value of STATIC_URL,
the leading slash is removed from that setting (now 'static/')
```

Source : https://docs.djangoproject.com/en/dev/releases/4.0/#deprecated-features-4-0

File: vulture_os/vulture_os/settings.py (hunk 5 above)
Comment (ELDiablO59152) [reply vulture_os/vulture_os/settings.py:191]: It is deprecated and throw a warning on django check routine : https://github.com/adamchainz/django-upgrade#use_l10n

File: vulture_os/vulture_os/settings.py (hunk 6 above)
Comment (ELDiablO59152) [reply vulture_os/vulture_os/settings.py:197]: It is

File: vulture_os/gui/urls.py (hunk 1 above)
Comment (ELDiablO59152) [reply vulture_os/gui/urls.py:40]: That's not mandatory, it was performed by django-upgrade and seemed relevant for me.
I can revert this change if needed.

Source : https://github.com/adamchainz/django-upgrade#urls

File: CHANGELOG [hunk 7]
```diff
@@ -10,12 +10,39 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ### Fixed
 - [PORTAL] [IDP] Refresh token validation with multiple IDP
 - [FRONTEND] Disable log condition if log forwarder disabled
 - [REPUTATION_CTX] Invalid headers templating
 - [PORTAL] [SSO] Avoid autologon password retrieval failures
 - [API_PARSER] [TRENDMICRO_VISIONONE] Fix Alerts endpoint params
+- [SETTINGS] Update CSRF_TRUSTED_ORIGINS format for Django 4
+- [URLS] Simplification of different paths
+- [DEPRECATION] Change request.META to request.headers
+- [DEPRECATION] get_random_string now requieres length argument
```
[the remark this answers is not in the document]
Comment (ELDiablO59152) [reply CHANGELOG:43]: I will correct that thanks <3

**Review by frikilax — APPROVED** [review]

**Pull request merged**